src/main/java/io/cdap/plugin/sap/odata/odata2/OData2Client.java

```diff
@@ -0,0 +1,165 @@
+   *                 (e.g., http://eccsvrname:8000/sap/opu/odata/sap/zgw100_dd02l_so_srv/).
+   * @param username username for basic authentication.
+   * @param password password for basic authentication.
+   */
+  public OData2Client(String rootUrl, String username, String password) {
+    super(rootUrl, username, password);
+  }
+
+  /**
+   * Get OData service metadata.
+   *
+   * @return OData service metadata.
+   * @throws ODataException if the metadata cannot be fetched.
+   */
+  public Edm getMetadata() {
+    if (metadata == null) {
+      initMetadata();
+    }
+    return metadata;
+  }
+
+  @Override
+  public Iterator<ODataEntity> queryEntitySet(String entitySetName, @Nullable String query) {
+    Edm metadata = getMetadata();
+    URI queryURI = getQueryURI(entitySetName, query);
+    HttpURLConnection connection = connect(queryURI.toASCIIString(), MediaType.APPLICATION_ATOM_XML);
+    try {
+      EdmEntitySet entitySet = metadata.getDefaultEntityContainer().getEntitySet(entitySetName);
+      InputStream content = (InputStream) connection.getContent();
+      ODataFeed feed = EntityProvider.readFeed(connection.getContentType(), entitySet, content, READ_PROPERTIES);
+
+      return new OData2EntityIterator(feed.getEntries().iterator());
+    } catch (IOException | EdmException | EntityProviderException e) {
+      throw new ODataException(String.format("Unable to read '%s' entity set.", entitySetName), e);
+    } finally {
+      // will close the content InputStream
+      connection.disconnect();
+    }
+  }
+
+  @Override
+  public EntityType getEntitySetType(String entitySetName) {
+    try {
+      EdmEntitySet entitySet = getMetadata().getDefaultEntityContainer().getEntitySet(entitySetName);
+      EdmEntityType edmEntityType = entitySet.getEntityType();
+      List<PropertyMetadata> properties = new ArrayList<>();
+      for (String propertyName : edmEntityType.getPropertyNames()) {
+        EdmProperty property = (EdmProperty) edmEntityType.getProperty(propertyName);
+        properties.add(edmToProperty(property));
+      }
+
+      return new EntityType(edmEntityType.getName(), properties);
+    } catch (EdmException e) {
+      throw new ODataException("Unable to get entity set type: " + e.getMessage(), e);
+    }
+  }
+
+  private PropertyMetadata edmToProperty(EdmProperty property) throws EdmException {
+    String type = property.getType().getName();
+    boolean nullable = property.getFacets().isNullable();
+    Integer precision = property.getFacets().getPrecision();
+    Integer scale = property.getFacets().getScale();
+
+    return new PropertyMetadata(property.getName(), type, nullable, precision, scale, null);
+  }
+
+  private void initMetadata() {
+    HttpURLConnection connection = connect(getMetadataURI().toASCIIString(), MediaType.APPLICATION_XML);
+    try {
+      InputStream content = connection.getInputStream();
```

Comment:
similar comment about this InputStream

Reply:
Added `content.close();` method invokation.